### PR TITLE
 Porting guide, Updated dumper and Script to find the address

### PR DESCRIPTION
**Pull request by PaulJenkin**
1. Porting guide is updated
2. for dumper i have added the print function which uses php as server side to record the value in a sync order
3. Added py script for finding the GOT base address
4. Added py script for finding the base address jumps
5.  Added py script for finding the address for rop

### CODE REVIEW

**Comment by sleirsgoevy** [comment]
Finder scripts do make sense (although for GOT offset I wouldn't automate this route, and will probably rewrite it; module splitter seems fine).

A PHP-based server does not make sense, as it introduces an unwanted dependency and a lot of trash in the `dumpers/` directory. I have a private solution for this, and I think uploading that instead will make more sense.

File: addForRop.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+				ropValue["pthread_create_addr"]=hex(int("0x{}".format(flData["libkernel.elf"][m-n].strip().split(":")[0]),16))
+				break
+			n+=1
+	if ln.count("syscall") and ropValue["get_errno_addr_addr"]==""  :
+		p=1
+		sub1=""
+		sub2=""
+		while p+1 < len(flData["libkernel.elf"]):
+			if flData["libkernel.elf"][m+p].count("jb"):
+				ropValue["get_errno_addr_addr"]=flData["libkernel.elf"][m+p].split("     ")[-1].split(" ")[0]
```
Comment (sleirsgoevy) [inline addForRop.py]: Is this nonsense ever used?

Comment (PaulJenkin) [reply addForRop.py]: I am not sure about your question

Line 66 is not used any where i have removed it.
I have tested this addForRop.py with the 6.72 dump and all the other dumps which i made for 6.20, seems to be working fine

Comment (sleirsgoevy) [reply addForRop.py]: I meant exactly that.

**Comment by sleirsgoevy** [comment]
I'll now merge this into a [separate branch](https://github.com/sleirsgoevy/bad_hoist/tree/porting-guide) to do some cleanup (e.g. remove this PHP crap) before it lands into master. Feel free to reopen it if you have something to add.

**Comment by sleirsgoevy** [comment]
Merged into master.

**Pull request merged**